docs/src/test-api/class-testinfo.md

```diff
@@ -38,33 +38,9 @@ Learn more about [test annotations](./test-annotations.md).
   - `path` <[void]|[string]> Optional path on the filesystem to the attached file.
   - `body` <[void]|[Buffer]> Optional attachment body used instead of a file.
 
-The list of files or buffers attached to the current test. Some reporters show test attachments. For example, you can attach a screenshot to the test.
+The list of files or buffers attached to the current test. Some reporters show test attachments.
 
-```js js-flavor=js
-const { test, expect } = require('@playwright/test');
-
-test('basic test', async ({ page }, testInfo) => {
-  await page.goto('https://playwright.dev');
-
-  // Capture a screenshot and attach it.
-  const path = testInfo.outputPath('screenshot.png');
-  await page.screenshot({ path });
-  testInfo.attachments.push({ name: 'screenshot', path, contentType: 'image/png' });
-});
-```
-
-```js js-flavor=ts
-import { test, expect } from '@playwright/test';
-
-test('basic test', async ({ page }, testInfo) => {
-  await page.goto('https://playwright.dev');
-
-  // Capture a screenshot and attach it.
-  const path = testInfo.outputPath('screenshot.png');
-  await page.screenshot({ path });
-  testInfo.attachments.push({ name: 'screenshot', path, contentType: 'image/png' });
-});
-```
+To safely add an attachment to a test, please use [`method: Test.attach`] instead of directly pushing onto this array. Future versions of Playwright Test may mark this array as read-only.
```

Comment:
Here are my thoughts:

* Encouraging a canonical/uniform way of doing something in a lib is better than having multiple ways (and needing to debug multiple ways), so I think it would be best to move people onto the explicit method instead of magically pushing onto an array
* Pushing onto array as a user feels a little haphazard and like I'm doing something I shouldn't be doing with Playwright.
* Pushing onto array removes any opportunity for validation early (the whole point of this MR), so future-looking, it would be best to force things through the explicit method whose internal implementation, validation, and data-structure can change without breaking code

I'm trying to think of cases where I'd ever want to push onto the array directly as a user instead of using the async method and cannot. 

However, I don't want to include more words in the docs than necessary/true, so if TestInfo.attachments.push continues to be in the docs and encouraged, let's drop this note. Thoughts?

Reply:
We try to never break our APIs, so it's unlikely that it will become readonly, unless there is a **very** strong reason for it. But it is safe to change all usages of `testInfo.attachments` in the docs to `testInfo.attach()`.